src/burp/restore.epp

```diff
@@ -11625,6 +11610,52 @@ void fix_system_generators(BurpGlobals* tdgbl)
 	}
 }
 
+void set_transaction(BurpGlobals* tdgbl)
+{
+/**************************************
+ *
+ *	s e t _ t r a n s a c t i o n
+ *
+ **************************************
+ *
+ * Functional description
+ *	Start a transaction with options
+ *	supported by the target server.
+ *
+ **************************************/
+
+	while (true)
+	{
+		if (tdgbl->gbl_use_auto_release_temp_blobid && tdgbl->gbl_use_no_auto_undo)
+		{
+			EXEC SQL SET TRANSACTION NO_AUTO_UNDO AUTO_RELEASE_TEMP_BLOBID;
+			if (gds_status->hasData())
```

Comment:
I understand you follow existing (old) code, but check for `isc_bad_tpb_form` makes this code more robust:

`
if (gds_status->hasData() && fb_utils::containsErrorCode(gds_status->getErrors(), isc_bad_tpb_form))
`